ets-demo/etstool.py

```diff
@@ -311,6 +336,62 @@ def flake8(runtime, toolkit, environment):
     execute(commands, parameters)
 
 
+@cli.command(name="generate-bundles")
+def generate_bundles():
+    """ Generate application bundles for each platform to be uploaded to
+    enthought/free repository
+    """
+
+    for platform in PLATFORMS:
+        bundle_dir = os.path.join("bundle", platform)
+        if not os.path.exists(bundle_dir):
+            os.makedirs(bundle_dir)
+
+        bundle_file = os.path.join(
+            bundle_dir,
+            f"etsdemo_{platform.replace('-', '_')}-{APP_BUNDLE_VERSION}.bundle"
+        )
+    
+        click.echo("Generating bundle {}".format(bundle_file))
+
+        edm_command = [
+            "edm",
+            "bundle",
+            "generate",
+            "--bundle-format",
+            "2.0",
+            "--platform",
+            platform,
+            "--version=3.6",
+            "--output-file",
+            bundle_file,
+        ] + sorted(full_app_dependencies)
+        try:
+            subprocess.check_call(edm_command)
+        except subprocess.CalledProcessError:
+            click.echo("Failed to generate bundle {}".format(bundle_file))
+
+
+@cli.command(name="upload-bundles")
+@click.option('--repository', default=DEFAULT_REPOSITORY)
+def upload_bundles():
+    """ Upload the generated bundles """
+    BUNDLES = []
+    for platform in PLATFORMS:
+        bundle_dir = os.path.join("bundle", platform)
+        bundle_file = os.path.join(
+            bundle_dir,
+            f"etsdemo_{platform.replace('-', '_')}-{APP_BUNDLE_VERSION}.bundle"
+        )
+        if os.path.exists(bundle_file):
+            print(bundle_file)
+            BUNDLES.append((bundle_file, platform))
+
+    for bundle, platform in BUNDLES:
+        cmd = "hatcher bundles upload enthought ets {platform} {bundle}"
```

Comment:
The docs say that the templaate for this command is `hatcher bundles upload <organization-name> <repository-name> <platform> <filename>`.  I'm unsure if `<filename>` includes the path or not.  
The code currently has `{bundle}` which is for example `bundle/rh7-x86_64/etsdemo_rh7_x86_64-1.0-1.bundle`.  I think this will likely need fixing, but I don't know if the hatcher command is supposed to be run from the sam directory as the bundle files?

Reply:
We are running the `etstool.py` script - which then uploads the bundles via the command - and the `bundle` cmdline argument needs to be a full path.